R/helper_starprep.R

```diff
@@ -249,10 +249,22 @@ starprep <- function(...,
                      se_type = NULL,
                      clusters = NULL,
                      alpha = 0.05) {
+
+  fits <- list(...)
+
+  first_object <- fits[[1]]
+
+  gave_list <- length(fits) == 1 & is.list(first_object)
```

Comment:
I think it might actually better to check if the first dots is an lm or not, instead of trying to write logic to detect a list - see also https://github.com/DeclareDesign/DeclareDesign/blob/3965a5c56631ecec2fc6ad05be28a7a196213b31/R/diagnose_design.R#L83 in DeclareDesign where there's ambiguity / similar logic bw data.frames and lists of data frames.

Reply:
I think it's better to cast as fits first so that we can sanitize input. I'll PR some changes in a second.

Reply:
The main problem is that lms and lm_robusts are also lists, right? I'll PR a change.